### PR TITLE
Added HitBTC and Binance exchange rate sources

### PR DESCRIPTION
**Pull request by einsteinx2**
- Added unit tests
- Cleaned up other unit tests

### CODE REVIEW

File: Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift [hunk 1]
```diff
@@ -204,6 +204,91 @@ public struct ExchangeRateParsing {
         return (exchangeRates, nil)
     }
     
+    public static func hitbtc(responseData: Data) -> ([ExchangeRate], BalanceError?) {
```
Comment (wolffan) [inline Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: i'm pretty sure we should start taking out the parsers for every exchange out of the exchangeratehandler file...

File: Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift [hunk 2]
```diff
@@ -204,6 +204,91 @@ public struct ExchangeRateParsing {
...
+        
+        // Verify the response is correct
+        guard let bodyOptional = try? responseString?.jsonDecode() as? [[String: Any]], let body = bodyOptional else {
+            return ([], .jsonDecoding)
+        }
+        
+        // Check for success
+        guard body.count > 0 else {
+            Log.error(message: "Did not return any prices")
+            return ([], .jsonDecoding)
```
Comment (wolffan) [inline Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: probably a new error, it's not .jsonDecoding but the fact that for whatever reason there's no data (so it's not our problem but theirs?)

File: Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift [hunk 3]
```diff
@@ -204,6 +204,91 @@ public struct ExchangeRateParsing {
...
+        }
+        
+        // Parse the exchange rates
+        var exchangeRates = [ExchangeRate]()
+        for rateDict in body {
+            guard let priceString = rateDict["price"] as? String, let price = Double(priceString), let symbol = rateDict["symbol"] as? String else {
+                return ([], .jsonDecoding)
```
Comment (wolffan) [inline Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: again this is probably not a jsonDecoding problem, but another one... it's been dificult do debug so more appropriate errors would help :)

File: Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift (hunk 1 above)
Comment (einsteinx2) [reply Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: Agreed, let's make this a ticket for 1.0.1

**Pull request merged**